autoload/OmniSharp/actions/members.vim

```diff
@@ -18,18 +18,75 @@ function! s:StdioFind(Callback) abort
   let opts = {
   \ 'ResponseHandler': function('s:StdioFindRH', [a:Callback])
   \}
-  call OmniSharp#stdio#Request('/currentfilemembersasflat', opts)
+  call OmniSharp#stdio#Request('/v2/codestructure', opts)
 endfunction
 
 function! s:StdioFindRH(Callback, response) abort
   if !a:response.Success | return | endif
-  call a:Callback(OmniSharp#locations#Parse(a:response.Body))
+  call a:Callback(s:ParseCodeStructure(a:response.Body))
+endfunction
+
+function! s:ParseCodeStructure(responseBody) abort
+  let locations = []
+  let filename = expand('%:p')
+  for element in a:responseBody.Elements
+    call s:ParseCodeStructureItemRec(element, filename, locations)
+  endfor
+  return locations
+endfunction
+
+function! s:ParseCodeStructureItemRec(item, filename, locations) abort
+  call add(a:locations, s:ParseCodeStructureItem(a:item, a:filename))
+  let children = get(a:item, 'Children', [])
+  if type(children) == type([])
+    for child in children
+      call s:ParseCodeStructureItemRec(child, a:filename, a:locations)
+    endfor
+  endif
+endfunction
+
+function! s:ParseCodeStructureItem(item, filename) abort
+  return {
+    \ 'filename': a:filename,
+    \ 'lnum':     a:item.Ranges.name.Start.Line,
+    \ 'col':      a:item.Ranges.name.Start.Column,
+    \ 'end_lnum': a:item.Ranges.name.End.Line,
+    \ 'end_col':  a:item.Ranges.name.End.Column - 1,
+    \ 'text':     s:ComputeItemSignature(a:item),
+    \ 'vcol': 1
+  \}
+endfunction
+
+function! s:ComputeItemSignature(item) abort
+  if type(a:item.Properties) != type({})
+    return get(a:item, 'Kind', '') . ' ' . a:item.DisplayName
+  endif
+  let line   = a:item.Ranges.name.Start.Line
+  let endcol = a:item.Ranges.name.Start.Column - 2
+  let textBeforeDisplayName = trim(getline(line)[:endcol], " \t", 1)
```

Comment:
I believe the 2nd and 3rd arguments to `trim()` are quite recent, and neovim errors on this line. Let's just use a substitute here:

```vim
  let textBeforeDisplayName = substitute(getline(line)[:endcol], '^\s*', '', '')
```

Reply:
agreed